src/Spanner/Database.php

```diff
@@ -0,0 +1,1313 @@
+     * find a record matching the Primary Key, and will replace the entire row.
+     * If a matching row is not found, it will be inserted.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $database->replaceBatch('Posts', [
+     *     [
+     *         'ID' => 1337,
+     *         'postTitle' => 'Hello World!',
+     *         'postContent' => 'Thanks for visiting our site!'
+     *     ], [
+     *         'ID' => 1338,
+     *         'postTitle' => 'Our History',
+     *         'postContent' => 'A little bit about us!'
+     *     ]
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param array $dataSet The row data to replace.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function replaceBatch($table, array $dataSet, array $options = [])
+    {
+        $mutations = [];
+        foreach ($dataSet as $data) {
+            $mutations[] = $this->operation->mutation(Operation::OP_REPLACE, $table, $data);
+        }
+
+        $session = $this->selectSession(SessionPoolInterface::CONTEXT_READWRITE);
+
+        $options['singleUseTransaction'] = $this->configureTransactionOptions();
+
+        try {
+            return $this->operation->commit($session, $mutations, $options);
+        } finally {
+            $session->setExpiration();
+        }
+    }
+
+    /**
+     * Delete one or more rows.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $keySet = new KeySet([
+     *     'keys' => [
+     *         1337, 1338
+     *     ]
+     * ]);
+     *
+     * $database->delete('Posts', $keySet);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param KeySet $keySet The KeySet to identify rows to delete.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function delete($table, KeySet $keySet, array $options = [])
+    {
+        $mutations = [$this->operation->deleteMutation($table, $keySet)];
+
+        $session = $this->selectSession(SessionPoolInterface::CONTEXT_READWRITE);
+
+        $options['singleUseTransaction'] = $this->configureTransactionOptions();
+
+        try {
+            return $this->operation->commit($session, $mutations, $options);
+        } finally {
+            $session->setExpiration();
+        }
+    }
```

Comment:
I don't see methods corresponding to [writeAtleastOnce](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java#L54) in the Java client.
Those methods are useful for high throughput applications since they only require one rpc.